tendrl/commons/objects/node/atoms/valid_node_details/__init__.py

```diff
@@ -0,0 +1,71 @@
+import etcd
+
+from tendrl.commons import objects
+from tendrl.commons.objects import AtomExecutionFailedError
+from tendrl.commons.objects.node import Node
+
+
+class ValidNodeDetails(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(ValidNodeDetails, self).__init__(*args, **kwargs)
+
+    def run(self):
+        node_ids = self.parameters.get('Node[]')
+        if not node_ids or len(node_ids) == 0:
+            raise AtomExecutionFailedError("Node[] cannot be empty")
+
+        for node_id in node_ids:
+            # Check if node has the OS details populated
+            try:
+                NS._int.client.read("nodes/%s/Os" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have OS details populated"
+                )
+
+            # Check if node has the CPU details populated
+            try:
+                NS._int.client.read("nodes/%s/Cpu" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have CPU details populated"
+                )
+
+            # Check if node has the Memory populated
+            try:
+                NS._int.client.read("nodes/%s/Memory" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have Memory details populated"
+                )
+
+            # Check if node has the block devices populated
+            try:
+                NS._int.client.read("nodes/%s/BlockDevices" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have block device details populated"
+                )
+
+            # Check if node has the disks data populated
+            try:
+                disks = NS._int.client.read("nodes/%s/Disks" % node_id)
+                # If disks have no child nodes, error out
+                if len(disks._children) == 0:
```

Comment:
@shtripat its a bad practice to use private vars of third party deps (_children), moreover i did mention in my comment to use .leaves 

Also, you need to check if all items like Os, memory etc are not empty